aws/table_aws_ssm_document.go

```diff
@@ -293,13 +301,34 @@ func listAwsSSMDocuments(ctx context.Context, d *plugin.QueryData, _ *plugin.Hyd
 
 func getAwsSSMDocument(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
 
-	var name string
+	var arn string
 	if h.Item != nil {
-		name = documentName(h.Item)
+		data, err := getgetAwsSSMDocumentArn(ctx, d, h)
+		if err != nil {
+		plugin.Logger(ctx).Error("aws_ssm_document.getAwsSSMDocument", "arn_formatting_error", err)
+			return nil, err
+		}
+		arn = data.(string)
 	} else {
-		name = d.EqualsQuals["name"].GetStringValue()
+		arn = d.EqualsQuals["arn"].GetStringValue()
+	}
+	metrixRegion := d.EqualsQualString(matrixKeyRegion)
+	arnSplit := strings.Split(arn, ":") // Split ARN to get the region
+
+	// Invalid arn check
+	if len(arnSplit) < 3 {
+		return nil, nil
+	}
+
+	// The same document name can be used in different regions.
+	// If a name is specified in the WHERE clause, the "get config" function will be executed. However, if the specified document name is available in different regions, the query will throw an error stating that the "get" call returned 23 results and the key column is not globally unique.
```

Comment:
I don't think this line is true anymore, since we don't have the `name` column as a key column for the GetConfig anymore